pylib/gyp/__init__.py

```diff
@@ -103,6 +103,14 @@ def Load(
     for (key, val) in generator.generator_default_variables.items():
         default_variables.setdefault(key, val)
 
+    output_dir = params["options"].generator_output or params["options"].toplevel_dir
+    if default_variables['GENERATOR'] == 'ninja':
+        default_variables.setdefault("PRODUCT_DIR_ABS", os.path.join(output_dir,
+                                     'out', default_variables['build_type']))
+    else:
+        default_variables.setdefault("PRODUCT_DIR_ABS", os.path.join(output_dir,
+                                     default_variables['build_type']))
+
```

Comment:
OPTIONAL: Readability...
```suggestion
    if default_variables['GENERATOR'] == 'ninja':
        prod_dir = os.path.join(output_dir, 'out', default_variables['build_type'])
    else:
        prod_dir = os.path.join(output_dir, default_variables['build_type'])
    default_variables.setdefault("PRODUCT_DIR_ABS", prod_dir)
```

Reply:
I've made these changes now, thanks.